twilight-util/src/builder/command.rs

```diff
@@ -506,9 +506,15 @@ impl IntegerBuilder {
         choice_name: &str,
         name_localizations: impl IntoIterator<Item = (K, V)>,
     ) -> Self {
-        let choice = self.0.choices.as_mut().expect("choices are set").iter_mut().find(
-            |choice| matches!(choice, CommandOptionChoice::Integer(CommandOptionChoiceData{ name, ..})  if name == choice_name),
-        );
+        let choice = self
+            .0
+            .choices
+            .as_mut()
+            .expect("choices are set")
+            .iter_mut()
+            .find(
+                |choice| matches!(choice, CommandOptionChoice{ name, value: CommandOptionChoiceValue::Integer(_), ..}  if name == choice_name),
```

Comment:
No need to match on the option choice value type. Also note that rustfmt might choke on this line due to it being too long.
```suggestion
                |choice| matches!(choice, CommandOptionChoice{ name, .. }  if name == choice_name),
```